nodejs/week2/src/backend/routes/meals.js

```diff
@@ -0,0 +1,123 @@
+const express = require("express");
+const dataAccess = require("../dataAccess.js");
+const { response } = require("express");
+const router = express.Router();
+const meals = "/data/meals.json";
+const reviews = "/data/reviews.json";
+
+///meals	Respond with the json for all the meals with reviews
+/*
+router.get("/meals", (request, response) => {
+  let mealsData = dataAccess.getDataFromJson(meals);
+  let reviewsData = dataAccess.getDataFromJson(reviews);
+  let mealsWithReviews = mealsData.map((meal) => {
+    meal.review = reviewsData.filter((review) => review.id === meal.id);
+    return meal;
+  });
+  response.send(mealsWithReviews);
+});*/
+
+///meals	Respond with the json for all the meals
+function getAllMeals(meals) {
+  let mealsData = dataAccess.getDataFromJson(meals);
+  result = mealsData;
+  return result;
+}
+
+///meals/{id}	Respond with the json for the meal with the corresponding id
+router.get("/meals/:id", (request, response) => {
+  let mealsData = dataAccess.getDataFromJson(meals);
+  let mealsIdString = request.params.id;
+  let mealsId = parseInt(mealsIdString);
+
+  const mealsIdOutput = mealsData.filter((item) => item.id === mealsId);
+  if (mealsIdOutput.length === 0) {
+    response.send("there are no meals with id: " + mealsIdString);
```

Comment:
In REST there's a formal way in which to tell a client that a resource with a given id doesn't exist.

Try to see how to do that with express: Send a 404 resource not found response

Reply:
Same in other routers